src/adapters/plugin/adapter-manager-proxy.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Proxy version of AdapterManager used by plugins.
+ *
+ * @module AdapterManagerProxy
+ */
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+const DEBUG = false;
+
+class AdapterManagerProxy extends EventEmitter {
+
+  constructor (pluginClient) {
+    super();
+
+    this.pluginClient = pluginClient;
+
+    this.on('property-changed', (property) => {
+      DEBUG && console.log('proxy: Got "property-changed" notification for',
+                           property.name, 'value:', property.value);
+
+      this.pluginClient.sendNotification('propertyChanged', {
```

Comment:
It would be good to have all constant strings like this defined somewhere. I think the best approach would be to change Constants.PROPERTY_CHANGED to be 'propertyChanged' then use it everywhere